### DISCUSSION TITLE
Please someone come up with a better route file system

### DISCUSSION THREAD

**Discussion by TadasV1** [sveltejs/kit#8786] [General]
I just started experimenting. At the beginner level all is good so far. 

But... this +page, +layout and so on system. **It takes out half the joy.** 

It creates vastly unnecessary cognitive load. Especially for newcomers. 

I read about the reasoning. But idk routes is a consistently used information piece. So it means you will always have this frustration. No matter what mental trick you gonna use. Because you always need to reference it in mind:

FILENAMES
**[[[ +page.server.js => /somedirectory ]]]**    ---------->  /somedirectory/+page.server.js
+page.svelte => /somedirectory         ----------> /somedirectory/+page.svelte
vs
**[[[ /_somedirectory.svelte ]]]**                          ----------> /somedirectory/_somedirectory.svelte
/_somedirectory.server.js                      ----------> /somedirectory/_somedirectory.server.js

[[[ info to keep in memory ]]]

Always need to keep **in working memory** what directory I'm working because these filename are absolutely useless. And afak working memory in programming is _prime real estate_. 

PLUS SIGN
And plus sign is very intrusive. Its way more pixels than underscore. and therefore more obstructive.

My point is - its all about cognitive load of the feature that is FREQUENTLY used.

**Comment by cnshsliu** [comment]
Plus, with a multi-tab editor opening several pages,  all tab labels are the same '+page.svelte' ... sigh

**Reply by cnshsliu** [reply]
Maybe treat filename which is the same as route folder name is a possible solution,
instead: 
`
routeA/+page.svelte
routeA/+page.ts
`
we may have:
`
routeA/routeA.svelte
routeA/routeA.ts
routeA/otherComponent.svelte
`

**Comment by teemingc** [comment]
One workaround, if you're using VSCode, is to change the setting `workbench.editor.labelFormat` from `default` to `short`.
This appends the directory name to the working file name in your editor, such as `routeA/+page.svelte`.

**Reply by TadasV1** [reply]
I do not use VSC. Tried did not like it.

**Reply by teemingc** [reply]
> I do not use VSC. Tried did not like it.

What kind of editor are using?
Does it have an option to display the current filepath?

**Reply by cnshsliu** [reply]
Thanks. I am using NeoVim as my daily editor. Normally I keep opening many tabs, I already configure NeoVim display shorten path as tab label.  but still looks clumsy since all base names are same '+page.svelte'.

![preview here](https://i.postimg.cc/1XFvL3s4/image.png)

I could configure NeoVim to use filepath as tab label, however, that's not a good idea since I have to keep the base name, like '+page.svelte',  because I may edit several files within the same folder together at the same time. If I use filepath as tab label,  the tab name of these files within the same folder will be all the same.

**Comment by fehnomenal** [comment]
I recently read saw this article in "What's new in Svelte": https://dev.to/devpunk/how-to-make-declarativecode-based-router-instead-of-file-based-router-in-sveltekit-2-3dd4

I won't try it but you could give it a shot.

**Reply by geoffrich** [reply]
I would not recommend this, because it involves modifying internal SvelteKit source code. You will have to maintain this patch every time you update SvelteKit.

There is no guarantee that the internal `routes` object will stay the same, because it is not exposed publicly.

**Reply by fehnomenal** [reply]
Well, I wouldn't recommend it either. I guess changing such a central piece of data deep down asks for trouble especially if internal changes result in changes to the values of the route parameters.

Just thought I would share it :shrug:
